### ISSUE TITLE
Error when scaling Replica Set

### ISSUE THREAD

**Issue by GayathriRameshK** [mongodb/mongodb-kubernetes-operator#1036]
**What did you do to encounter the bug?**
Deploy replica set with members:1 (with tls and replicaset horizons enabled)
Scale up replica set to members:3 (with tls and replicaset horizons enabled)
Scale down replica set to members:1 (with tls and replicaset horizons enabled)

**What did you expect?**
Successfully running 3 mongo pods 
When scale down to 1 , 1 mongodb pods should be running successfully

**What happened instead?**
mdbc is getting into failed state.
Mongo pod status as below
upgrade-mongodb-0                             1/2     Running     0          13h
when scale down, operator pod is getting crashed and mdbc is getting into pending state and replicaset is not recoverable(scale down issue is predominent with operator version 0.7.2, 0.7.3, 0.7.4)

**Operator Information**
 - Operator Version - 0.7.4
 - MongoDB Image used - 4.2.12, 5.0.6

If possible, please include:
 - `kubectl describe` output
$Kubectl describe pod upgrade-mongodb-0
Events:
  Type     Reason     Age                    From     Message
  ----     ------     ----                   ----     -------
  Warning  Unhealthy  4m5s (x4824 over 13h)  kubelet  Readiness probe failed:

 - yaml definitions for your objects
```
#use for testing
---
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  name: upgrade-mongodb
spec:
  members: 1
  type: ReplicaSet
  version: "4.2.12"
  security:
    authentication:
      modes: ["SCRAM"]
      #ignoreUnknownUsers: true
#custom role start
    roles:
      - role: role-restore
        db: admin
        privileges:
          - resource:
              anyResource: true
            actions:
              - anyAction
        roles: []
#custom role end
  # Enable TLS certificate - start
    tls:
      enabled: true
      certificateKeySecretRef:
        name: upgrade-sec-mongodb
      caConfigMapRef:
        name: upgrade-ca-mongodb
# Enable TLS certificate - end
#additional parameter in mongod config file - start
  additionalMongodConfig:
    net.tls.mode: requireTLS
    storage.directoryPerDB: true
#additional parameter in mongod config file - end
  replicaSetHorizons: 
    - "horizons": "<hostname>:<port>"
    - "horizons": "<hostname>:<port>"
    - "horizons": "<hostname>:<port>"

  users:
    - name: admin
      db: admin
      passwordSecretRef: # a reference to the secret that will be used to generate the user's password
        name: dbauser
      roles:
        - name: root
          db: admin
        - name: role-restore
          db: admin
      scramCredentialsSecretName: mongo-scram



#use specific storage class, CPU & Memory - start
  # persistent: true
  statefulSet:
    spec:
      volumeClaimTemplates:
        - metadata:
            name: data-volume
          spec:
            accessModes: [ "ReadWriteOnce" ]
            storageClassName:  #storage class need to be changed according to the infra
            resources:
              requests:
                storage: 6Gi
        - metadata:
            name: logs-volume
          spec:
            accessModes: [ "ReadWriteOnce" ]
            storageClassName:  #storage class need to be changed according to the infra
            resources:
              requests:
                storage: 3Gi

      template:
        spec:
          containers:
            - name: "mongodb-agent"
              readinessProbe:
                timeoutSeconds: 10 
              resources:
                requests:
                  cpu: 200m
                  memory: 200Mi
                limits:
                  cpu: 250m
                  memory: 250Mi
            - name: "mongod"
              resources:
                requests:
                  cpu: 500m #500m
                  memory: 500Mi #500Mi
                limits:
                  cpu: 1
                  memory: 1Gi
            
#use specific storage class, CPU & Memory - end
```
 - log files for the operator and database pods
operator logs:
 DEBUG   agent/agent_readiness.go:105    The Agent in the Pod 'upgrade-mongodb-0' hasn't reached the goal state yet (goal: 2, agent: 1)        {"ReplicaSet": "mongo-upgrade/upgrade-mongodb"}
  DEBUG   agent/agent_readiness.go:101    The Pod '' doesn't have annotation 'agent.mongodb.com/version' yet    {"ReplicaSet": "mongo-upgrade/upgrade-mongodb"}

mongodb-agent logs:

[.info] [src/director/director.go:planAndExecute:563] <upgrade-mongodb-0>  Step=WaitRsConfCommitted as part of Move=WaitRsConfCommitted in plan failed : <upgrade-mongodb-0> Postcondition not yet met for step WaitRsConfCommitted because 
['currentState.ReplSetConf.Members' != 'desiredState.ReplSetConf.Members' :

**Comment by GayathriRameshK** [comment]
can anyone give updates on this issue?

**Comment by dan-mckean** [comment]
Hi, I'm Dan and I'm the Product Manager for MongoDB's support of Kubernetes.

I'm doing some work right now to try and identify how the Community Operator is being used. The Community Operator is something I inherited when I started at MongoDB, but it doesn't get as much attention from us as we'd like and we're trying to understand how it's used in order to establish its future. It will help us prioritize future issues and PRs raised by the community 🙂

Here's a super short survey (it's much easier for us to review all the feedback that way!): [https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link](https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link)

 If you'd rather email me instead: [dan.mckean@mongodb.com](mailto:dan.mckean@mongodb.com?subject=MongoDB%20Community%20Operator%20feedback)

Thank you in advance!
Dan
